Controller/RecommendationController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * File containing the RecommendationController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\RecommendationBundle\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use EzSystems\RecommendationBundle\Client\YooChooseNotifier;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+
+class RecommendationController extends Controller
+{
+    public function recommendationsAction( Request $request )
+    {
+        if ( !$request->isXmlHttpRequest() ) {
+            throw new BadRequestHttpException();
+        }
+
+        $userId = $this->getRepository()->getCurrentUser()->id;
+        $locationId = $request->query->get( 'locationId' );
+        $limit = $request->query->get( 'limit' );
+        $scenarioId = $request->query->get( 'scenarioId' );
+
+        $notifier = $this->get( 'ez_recommendation.client.yoochoose_notifier' );
```

Comment:
Would you consider making your controller a [controller as a service](http://symfony.com/doc/current/cookbook/controller/service.html) ? It's really sexier (and cleaner, but who cares) :-)